### PR TITLE
DENG-6779-1: Help Page Link

### PR DESCRIPTION
**Pull request by jsu**
# Summary [DENG-6779](https://openmail.atlassian.net/browse/DENG-6779)

1. Need to expose the help page and alert target to internal alert messages.
2. Also update some stuff that have been bothering me. 

# Deployment Instructions

1. Add 2 text columns: `data_alert_metadata.link` and `data_alert_metadata.affected_link`.
2. build_and_deploy_airflow
3. Add `link` in data_delay_tagging sheets.

### CODE REVIEW

File: airflow_prometheus_exporter/prometheus_exporter.py [hunk 1]
```diff
@@ -15,13 +15,13 @@
...
 from sqlalchemy import Column, String, and_, func
 from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy_utcdatetime import UTCDateTime
 
 from airflow.configuration import conf
 from airflow.models import DagModel, DagRun, TaskFail, TaskInstance, XCom
 from airflow.plugins_manager import AirflowPlugin
 from airflow.settings import RBAC, Session
 from airflow.utils.log.logging_mixin import LoggingMixin
-from airflow.utils.sqlalchemy import UtcDateTime
```
Comment (jsu) [inline airflow_prometheus_exporter/prometheus_exporter.py:24]: It's for local dev purpose. So we don't have to include airflow utils but sqlalchemy_utcdatetime.

File: airflow_prometheus_exporter/prometheus_exporter.py [hunk 2]
```diff
@@ -46,19 +46,9 @@ def session_scope(session):
...
-        alert_report_classification = Column(String)
-        sla_interval = Column(String)
-        sla_time = Column(String)
-        group_pagerduty = Column(String)
-        group_business_line = Column(String)
-        inhibit_rule = Column(String)
-        latest_successful_run = Column(UtcDateTime)
+        __table_args__ = {"autoload": True}
+        dag_id = Column(String, primary_key=True)
+        latest_successful_run = Column(UTCDateTime)
```
Comment (jsu) [inline airflow_prometheus_exporter/prometheus_exporter.py:51]: Keep only the ones that need to be declared explicitly. Leave the rest autoload.

File: airflow_prometheus_exporter/prometheus_exporter.py [hunk 3]
```diff
@@ -389,7 +379,8 @@ def sla_check(sla_interval, sla_time, max_execution_date, cadence, execution_dat
         local_datetime = utc_datetime.astimezone(pytz.timezone(TIMEZONE_LA))
         sla_datetime = pytz.timezone(TIMEZONE_LA).localize(
             datetime.datetime.combine(
-                local_datetime.date(), datetime.time(int(sla_time))
+                local_datetime.date(),
+                datetime.datetime.strptime(sla_time, "%H:%M").time()
             )
```
Comment (jsu) [inline airflow_prometheus_exporter/prometheus_exporter.py:384]: An update to handle "21:30" in a better way.

File: airflow_prometheus_exporter/prometheus_exporter.py [hunk 4]
```diff
@@ -430,20 +421,21 @@ def get_sla_miss_dags():
...
+                DelayAlertMetaData.cadence,
+                DelayAlertMetaData.dag_id,
                 DelayAlertMetaData.group_business_line,
+                DelayAlertMetaData.group_pagerduty,
                 DelayAlertMetaData.inhibit_rule,
                 DelayAlertMetaData.latest_successful_run,
+                DelayAlertMetaData.link,
+                DelayAlertMetaData.severity,
+                DelayAlertMetaData.sla_interval,
+                DelayAlertMetaData.sla_time,
```
Comment (jsu) [inline airflow_prometheus_exporter/prometheus_exporter.py]: Added `link` and sort the fields.

**Comment by jsu** [comment]
## Test

```

# TYPE airflow_dags_sla_miss gauge
airflow_dags_sla_miss{alert_external_classification="missing",alert_report_classification="missing",alert_target="missing",dag_id="buyside_snapchat",group_business_line="missing",group_pagerduty="dataeng-alert-stage-b1",inhibit_rule="missing",link="link to page",severity="medium"} 0.0
# HELP airflow_tasks_sla_miss Airflow tasks missing the sla
# TYPE airflow_tasks_sla_miss gauge
airflow_tasks_sla_miss{alert_external_classification="missing",alert_report_classification="missing",alert_target="missing",dag_id="buyside_snapchat",group_business_line="missing",group_pagerduty="dataeng-alert-stage-b1",inhibit_rule="missing",link="link to page",severity="medium",task_id="start"} 0.0

```

Screen Shot 2021-03-31 at 08 22 47

**Review by megazero93 — APPROVED** [review]

**Review by bakersys1 — APPROVED** [review]

**Comment by jsu** [comment]
Screen Shot 2021-04-05 at 06 44 02

**Pull request merged**